tasks/main.yml

```diff
@@ -1,29 +1,50 @@
 - apt: pkg={{ item }} state=present update_cache=true
-  with_items: [nginx, python-pip]
+  with_items:
+    - nginx
+    - python-pip
   sudo: yes
 
 # Install our site
-- template: dest="/etc/nginx/sites-available/{{ app_name }}" src=nginx_api.j2 mode=0644
+- template:
+    dest=/etc/nginx/sites-available/{{ app_name }}
+    src=nginx_api.j2
+    mode=0644
   sudo: yes
 
-- file: dest="/etc/nginx/sites-enabled/{{ app_name }}" src="/etc/nginx/sites-available/{{ app_name }}" state=link
+- file:
+    dest=/etc/nginx/sites-enabled/{{ app_name }}
+    src=/etc/nginx/sites-available/{{ app_name }}
+    state=link
   sudo: yes
 
 # Remove the default site
-- file: path=/etc/nginx/sites-enabled/default state=absent
+- file:
+    path=/etc/nginx/sites-enabled/default
+    state=absent
   sudo: yes
 
 - command: /etc/init.d/nginx restart
   sudo: yes
 
 # Ensure that uwsgi is installed
-- command: "{{ app_bin_dir }}/pip install uwsgi==2.0.5.1"
+- command: "{{ app_bin_dir }}/pip install uwsgi=={{ uwsgi_version }}"
   sudo: yes
   sudo_user: "{{ app_user }}"
 
-# Do a restart of the supervisor task.
-# This is because ansible will run the dependency EDITD.supervisor_task before
-# this one! It probably won't run properly because some essential steps are in
-# this role, so we restart it.
-- command: "supervisorctl restart {{ app_name }}:*"
+# Make sure the supervisor task is indeed started. This is because ansible will
+# run the dependency role EDITD.supervisor_task before this one! It probably
+# won't run properly because some essential steps like installing the uwsgi
+# package are in this role, so we make sure it is started correctly.
+- supervisorctl:
+    name="{{ app_name }}:"
+    state=started
   sudo: yes
+
+# Optionally, restart the supervisor task group when requested by the user.
+# This does not gracefully restart the uwsgi workers, so not preferable,
+# but it might be needed for certain use cases.
+- supervisorctl:
+    name="{{ app_name }}:"
+    state=restarted
+  sudo: yes
+  when: restart_app
```

Comment:
Good point, I added this afterward and didn't link it back to the supervisor_task call.
How's this? 2e5e48d